plugins/flytekit-data-fsspec/flytekitplugins/fsspec/pandas.py

```diff
@@ -0,0 +1,69 @@
+import os
+import typing
+from pathlib import Path
+
+import pandas as pd
+from flytekitplugins.fsspec.persist import FSSpecPersistence, s3_setup_args
+
+from flytekit import FlyteContext
+from flytekit.configuration import aws as _aws_config
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.models.types import StructuredDatasetType
+from flytekit.types.structured.structured_dataset import (
+    PARQUET,
+    S3,
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+)
+
+
+def get_storage_options(uri: str) -> typing.Optional[typing.Dict]:
```

Comment:
you're sure this function does what you want right?  it feels kinda weird to me.  cuz it calls s3_setup_args but there's like some duplicate logic there.  as long as you're happy with it.   (there's some funniness in the unit test i didn't catch the first time, sorry about that.  see the comment there)

Reply:
just revised it